lib/urbanopt/scenario/default_reports/reporting_period.rb

```diff
@@ -35,15 +35,197 @@ module Scenario
+          existing_period.district_heating += new_period.district_heating
+          existing_period.water += new_period.water
+          existing_period.electricity_produced += new_period.electricity_produced
+          existing_period.heating += new_period.heating
+          existing_period.cooling += new_period.cooling
+          existing_period.interior_lighting += new_period.interior_lighting
+          existing_period.exterior_lighting += new_period.exterior_lighting
+          existing_period.interior_equipment += new_period.interior_equipment
+          existing_period.exterior_equipment += new_period.exterior_equipment
+          existing_period.fans += new_period.fans
+          existing_period.pumps += new_period.pumps
+          existing_period.heat_rejection += new_period.heat_rejection
+          existing_period.humidification += new_period.humidification
+          existing_period.heat_recovery += new_period.heat_recovery
+          existing_period.water_systems += new_period.water_systems
+          existing_period.refrigeration += new_period.refrigeration
+          existing_period.generators += new_period.generators
+          existing_period.photovoltaic += new_period.photovoltaic
+          existing_period.usage_cost += new_period.usage_cost
+          existing_period.demand_cost += new_period.demand_cost
+          existing_period.time_setpoint_not_met_during_occupied_cooling += new_period.time_setpoint_not_met_during_occupied_cooling
+          existing_period.time_setpoint_not_met_during_occupied_heating += new_period.time_setpoint_not_met_during_occupied_heating
+          existing_period.time_setpoint_not_met_during_occupied_heating += new_period.time_setpoint_not_met_during_occupied_heating
+          existing_period.time_setpoint_not_met_during_occupied_hours += new_period.time_setpoint_not_met_during_occupied_hours         
+                             
+          return existing_period
+        end
+        
+        def self.merge_reporting_periods(existing_periods, new_periods)
+                    
+          # TODO: match new periods to existing periods and call merge_reporting_period
+          #puts "existing_periods = #{existing_periods.name}"
+          #puts "new_periods = #{new_periods.name}"
+
+          id_list = []
+          id_list = existing_periods.collect {|x| x.id}
+      
+          new_periods.each do |x_new|
+              
+            if id_list.include?(x_new.id)
+                  
+              # when looping over the new_periods ids find the index of the id_list with the same id            
+              index = id_list.find_index(x_new.id)
+                    
+              # modify the existing_periods by merging the new periods
+              existing_periods[index] = merge_reporting_period(existing_periods[index], x_new)
+      
+            else
+              #insert the new hash in to the array 
+              existing_periods << x_new
```

Comment:
This might actually be a warning.  If existing_periods is empty (e.g. when merging the first feature_report) then `existing_periods  = new_periods`.  Otherwise, we expect all simulations to report for the same periods.